File: 6/src6/weeks/3/compare3.py
# Compares two strings for equality while checking (succinctly) for errors

import sys
from cs50 import get_string

# Get a string
s = get_string("s: ")
if not s:
    sys.exit(1)

# Get another string
t = get_string("t: ")
if not t:
    sys.exit(1)

# Compare strings for equality
if s == t:
    print("same")
else:
    print("different")

File: 4/src4/compare3.c
// Compares two strings for equality while checking (succinctly) for errors

#include <cs50.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // get a string
    char *s = get_string("s: ");
    if (!s)
    {
        return 1;
    }

    // get another string
    char *t = get_string("t: ");
    if (!t)
    {
        return 1;
    }

    // compare strings for equality
    if (strcmp(s, t) == 0)
    {
        printf("same\n");
    }
    else
    {
        printf("different\n");
    }
    return 0;
}
